homeassistant/components/goodwe/select.py

```diff
@@ -0,0 +1,78 @@
+"""GoodWe PV inverter selection settings entities."""
+import logging
+
+from goodwe import Inverter, InverterError
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.const import ENTITY_CATEGORY_CONFIG
+from homeassistant.helpers.entity import DeviceInfo
+
+from .const import DOMAIN, KEY_DEVICE_INFO, KEY_INVERTER
+
+_LOGGER = logging.getLogger(__name__)
+
+
+INVERTER_OPERATION_MODES = [
+    "General mode",
+    "Off grid mode",
+    "Backup mode",
+    "Eco mode",
+]
+
+OPERATION_MODE = SelectEntityDescription(
+    key="operation_mode",
+    name="Inverter operation mode",
+    icon="mdi:solar-power",
+    entity_category=ENTITY_CATEGORY_CONFIG,
+)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the inverter select entities from a config entry."""
+    inverter = hass.data[DOMAIN][config_entry.entry_id][KEY_INVERTER]
+    device_info = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE_INFO]
+
+    # read current operating mode from the inverter
+    try:
+        active_mode = await inverter.get_operation_mode()
+        if 0 <= active_mode < len(INVERTER_OPERATION_MODES):
```

Comment:
Please adjust here too and move the lines that can't raise.

Reply:
Done

Reply:
It's not changed yet. Is there a commit that's not pushed?

Reply:
Indeed, I forgot to push, my mistake.